File: dotnet/DesignPatterns/StructuralDesignPatterns/DecoratorDesignPattern/DecoratorDesignPattern.Example1/CharacterStatsDisplayHelper.cs
namespace DecoratorDesignPattern.Example1;

public static class CharacterStatsDisplayHelper
{
    public static void ShowCharacterStats(ICharacter character)
    {
        Console.WriteLine("============================================================");
        Console.WriteLine($"KARAKTER: {character.GetName()}");
        Console.WriteLine("============================================================");
        Console.WriteLine($"Açıklama: {character.GetDescription()}");
        Console.WriteLine($"Can      : {character.GetHealth()} HP");
        Console.WriteLine($"Saldırı  : {character.GetAttackPower()} DMG");
        Console.WriteLine($"Savunma  : {character.GetDefense()} DEF");
        Console.WriteLine($"Hız      : {character.GetSpeed()} SPD");

        Console.WriteLine("============================================================");
        Console.WriteLine();
    }
}

File: nodejs/DesignPatterns/StructuralDesignPatterns/DecoratorDesignPattern/DecoratorDesignPattern.Example1/src/CharacterStatsDisplayHelper.ts
import { ICharacter } from './ICharacter';

export function showCharacterStats(character: ICharacter): void {
  console.log('============================================================');
  console.log(`KARAKTER: ${character.getName()}`);
  console.log('============================================================');
  console.log(`Açıklama: ${character.getDescription()}`);
  console.log(`Can      : ${character.getHealth()} HP`);
  console.log(`Saldırı  : ${character.getAttackPower()} DMG`);
  console.log(`Savunma  : ${character.getDefense()} DEF`);
  console.log(`Hız      : ${character.getSpeed()} SPD`);
  console.log('============================================================');
  console.log();
}
